docs/s2i.md

```diff
@@ -0,0 +1,74 @@
+# s2i
+
+Containers Tekton Tasks
+
+---
+
+# `s2i` Tekton Task
+
+The `s2i` Task helps in building reproducible container images from source code i.e. task for supporting s2i functionality.
+
+The s2i Task has been customized with builder images specific to various languages and have been named appropriately as follows:
+
+| Task Name  | Builder Image Used                                                     |
+| ---------- | ---------------------------------------------------------------------- |
+| s2i-python | http://registry.access.redhat.com/ubi8/python-39:latest                |
+| s2i-go     | http://registry.access.redhat.com/ubi8/go-toolset:1.19.10-3            |
+| s2i-java   | http://registry.access.redhat.com/ubi8/openjdk-11:latest               |
+| s2i-dotnet | http://registry.access.redhat.com/ubi8/dotnet-60:6.0-37.20230802191230 |
+| s2i-php    | http://registry.access.redhat.com/ubi9/php-81:1-29                     |
+| s2i-nodejs | http://registry.access.redhat.com/ubi8/nodejs-18:latest                |
+| s2i-perl   | http://registry.access.redhat.com/ubi9/perl-532:1-91                   |
+| s2i-ruby   | http://registry.access.redhat.com/ubi9/ruby-31:1-50                    |
+
+In case, the above builder images associated with the languages aren’t satisfactory for your source code, you can change it using appropriate parameter.
+
+Please, consider the usage example below:
+
+```yaml
+---
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata: {}
+spec:
+  taskRef:
+    name: s2i-python
+  params:
+    - name: IMAGE
+      value: registry.registry.svc.cluster.local:32222/task-containers/task-s2i-python:latest
+```
+
+In case the Container Registry requires authentication, please consider the [Tekton Pipelines documentation][tektonPipelineAuth]. In a nutshell, you need to create a Kubernetes Secret describing the following attributes:
+
+```bash
+kubectl create secret docker-registry imagestreams \\
+  --docker-server="image-registry.openshift-image-registry.svc:5000" \\
+  --docker-username="${REGISTRY_USERNAME}" \\
+  --docker-password="${REGISTRY_TOKEN}"
+
+```
```

Comment:
Please use a single backslash (`\`) at the end of each line, and also, please remove the leftover line at the end of the block.